src/client/Microsoft.Identity.Client/PlatformsCommon/DeviceAuthHelper.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Net;
+using Microsoft.Identity.Client.Http;
+using Microsoft.Identity.Client.PlatformsCommon.Shared;
+using Microsoft.Identity.Client.Utils;
+
+namespace Microsoft.Identity.Client.PlatformsCommon
+{
+    internal class DeviceAuthHelper
+    {
+        public static IDictionary<string, string> ParseChallengeData(HttpResponse response)
+        {
+            IDictionary<string, string> data = new Dictionary<string, string>();
+            string wwwAuthenticate = response.Headers.GetValues(PKeyAuthConstants.WwwAuthenticateHeader).Single();
```

Comment:
Careful with `Single`, it will crash if there isn't exactly 1 header with that name. Consider using `SingleOrDefault` and then if wwwAuthenticate is null, return an empty dict.